src/vehicle.h

```diff
@@ -1148,7 +1148,7 @@ class vehicle : public JsonSerializer, public JsonDeserializer
     std::vector<int> speciality;       // List of parts that will not be on a vehicle very often, or which only one will be present
     std::vector<int> floating;         // List of parts that provide buoyancy to boats
     std::set<std::string> tags;        // Properties of the vehicle
-
+    std::map<itype_id,int> fuel_remainder; // Remaining fuel after consumption.
```

Comment:
I don't see any problems with the approach, but this is a change that should be commented on somewhere in the OP post.